Tests/UnitTests/Networking/Backend/BackendGetVirtualCurrenciesTests.swift

```diff
@@ -0,0 +1,291 @@
+        self.httpClient.mock(
+            requestPath: .getVirtualCurrencies(appUserID: Self.userID),
+            response: .init(statusCode: .success, response: Self.twoVirtualCurrenciesResponse)
+        )
+
+        let result = waitUntilValue { completed in
+            self.virtualCurrenciesAPI.getVirtualCurrencies(
+                appUserID: Self.userID,
+                isAppBackgrounded: false,
+                completion: completed
+            )
+        }
+
+        expect(result).to(beSuccess())
+        expect(self.httpClient.calls).to(haveCount(1))
+        expect(self.operationDispatcher.invokedDispatchOnWorkerThreadDelayParam) == JitterableDelay.none
+    }
+
+    // MARK: - Caching Tests
+    func testGetVirtualCurrenciesCachesForSameUserID() {
+        self.httpClient.mock(
+            requestPath: .getVirtualCurrencies(appUserID: Self.userID),
+            response: .init(statusCode: .success,
+                            response: Self.twoVirtualCurrenciesResponse,
+                            delay: .milliseconds(10))
+        )
+        self.virtualCurrenciesAPI.getVirtualCurrencies(
+            appUserID: Self.userID,
+            isAppBackgrounded: false
+        ) { _ in }
+
+        self.virtualCurrenciesAPI.getVirtualCurrencies(
+            appUserID: Self.userID,
+            isAppBackgrounded: false
+        ) { _ in }
+
+        expect(self.httpClient.calls).toEventually(haveCount(1))
+    }
+
+    func testGetVirtualCurrenciesDoesntCacheForMultipleUserID() {
+        let response = MockHTTPClient.Response(
+            statusCode: .success,
+            response: Self.twoVirtualCurrenciesResponse
+        )
+        let userID2 = "user_id_2"
+
+        self.httpClient.mock(requestPath: .getVirtualCurrencies(appUserID: Self.userID), response: response)
+        self.httpClient.mock(requestPath: .getVirtualCurrencies(appUserID: userID2), response: response)
+
+        self.virtualCurrenciesAPI.getVirtualCurrencies(
+            appUserID: Self.userID,
+            isAppBackgrounded: false
+        ) { _ in }
+        self.virtualCurrenciesAPI.getVirtualCurrencies(
+            appUserID: userID2,
+            isAppBackgrounded: false
+        ) { _ in }
+
+        expect(self.httpClient.calls).toEventually(haveCount(2))
+    }
+
+    // MARK: - Response Tests
+    func testGetVirtualCurrenciesTwoCurrencies() throws {
+        self.httpClient.mock(
+            requestPath: .getVirtualCurrencies(appUserID: Self.userID),
+            response: .init(statusCode: .success,
+                            response: Self.twoVirtualCurrenciesResponse,
+                            delay: .milliseconds(10))
+        )
+        let result: Atomic<Result<VirtualCurrenciesResponse, BackendError>?> = nil
+
+        self.virtualCurrenciesAPI.getVirtualCurrencies(
+            appUserID: Self.userID,
+            isAppBackgrounded: false
+        ) { result.value = $0 }
+
+        expect(result.value).toEventuallyNot(beNil())
```

Comment:
I think we should avoid the `toEventually` clauses (which can make the test flakier and slower). In this case, we can just wait for the result instead.
```suggestion
        let result = waitUntilValue { completed in
            self.virtualCurrenciesAPI.getVirtualCurrencies(
                appUserID: Self.userID,
                isAppBackgrounded: false,
                completion: completed
            )
        }
```

Line `expect(result.value).toEventuallyNot(beNil())` should not be needed now because there's already the `let response = try XCTUnwrap(result.value?.value)` check